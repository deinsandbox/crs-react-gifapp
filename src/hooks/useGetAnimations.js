import { useState, useEffect } from "react";

import { getAnimations } from "../helpers/animations";
import { wait } from "../helpers/wait";

const useGetAnimations = (category) => {
  const init = {
    data: [],
    isLoading: true,
  };
  const [state, setState] = useState(init);

  useEffect(() => {
    const response = getAnimations(category);
    response.then((images) => {
      const result = {
        data: images,
        isLoading: false,
      };

      wait(1, () => setState(result));
    });
  }, [category]);

  return state;
};

export default useGetAnimations;
