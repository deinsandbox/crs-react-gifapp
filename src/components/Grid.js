import React from "react";
import PropTypes from "prop-types";
import useGetAnimations from "../hooks/useGetAnimations";
import Item from "./Item";

const Grid = ({ category }) => {
  const { data: images, isLoading } = useGetAnimations(category);

  return (
    <>
      <div className="masonry-wrapper">
        <h3>{category}</h3>

        <div className="masonry">
          {isLoading && (
            <span className="animate__animated animate__flash">Loading...</span>
          )}

          {!isLoading &&
            images.length > 0 &&
            images.map((image) => {
              return <Item key={image.id} {...image} />;
            })}

          {!isLoading && images.length === 0 && <span>No result found</span>}
        </div>
      </div>
    </>
  );
};

Grid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default Grid;
