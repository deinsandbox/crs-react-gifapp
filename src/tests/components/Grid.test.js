import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import Grid from "../../components/Grid";

import useGetAnimations from "../../hooks/useGetAnimations";
jest.mock("../../hooks/useGetAnimations");

describe("Test on <Grid /> component", () => {
  const category = "test";
  const images = [
    {
      id: "01234",
      title: "name1",
      url: "http://localhost/image1.gif",
    },
    {
      id: "12345",
      title: "name2",
      url: "http://localhost/image2.gif",
    },
  ];

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    useGetAnimations.mockReturnValue({ data: [], loading: true });
  });

  test("should render the component correctly", () => {
    wrapper = shallow(<Grid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should show category value as title", () => {
    wrapper = shallow(<Grid category={category} />);
    const title = wrapper.find("h3");
    //console.log(JSON.stringify(title.text()));
    expect(title.text()).toBe(category);
  });

  test("Should show items when load images on useGetAnimations", () => {
    useGetAnimations.mockReturnValue({ data: images, loading: false });
    wrapper = shallow(<Grid category={category} />);

    //expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("Item").length).toBe(images.length);
  });
});
