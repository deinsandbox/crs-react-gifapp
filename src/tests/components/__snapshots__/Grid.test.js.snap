// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Grid /> component should render the component correctly 1`] = `
<Fragment>
  <div
    className="masonry-wrapper"
  >
    <h3>
      test
    </h3>
    <div
      className="masonry"
    >
      <span>
        No result found
      </span>
    </div>
  </div>
</Fragment>
`;
